src/plugins/rnnoise.vencordDesktop/index.tsx

```diff
@@ -0,0 +1,164 @@
+        </div>
+        <div className={cl("popout-desc")}>
+            Enable AI noise suppression! Make some noise&mdash;like becoming an air conditioner, or a laptop fan&mdash;while speaking. Your friends will hear nothing but your beautiful voice ✨
+        </div>
+    </div>;
+}
+
+export default definePlugin({
+    name: "AI Noise Suppression",
+    description: "Uses an open-source AI model (RNNoise) to remove background noise from your microphone",
+    authors: [Devs.Vap],
+    settings,
+
+    patches: [
+        {
+            find: "window.webkitAudioContext",
+            replacement: {
+                match: /(?<=\i\.acquire=function\((\i)\)\{return )navigator\.mediaDevices\.getUserMedia\(\1\)(?=\})/,
+                replace: m => `${m}.then(stream => $self.connectRnnoise(stream))`
+            },
+        },
+        {
+            find: "renderNoiseCancellation()",
+            replacement: {
+                match: /(?<=(\i)\.jsxs?.{0,70}children:\[)(?=\i\?\i\.renderNoiseCancellation\(\))/,
+                replace: (_, react) => `${react}.jsx($self.NoiseSupressionButton, {}),`
+            },
+        },
+        {
+            find: "Messages.USER_SETTINGS_NOISE_CANCELLATION_KRISP",
+            replacement: [{
+                match: /(?<=(\i)=\i\?\i\.KRISP:\i.{1,20}?;)/,
+                replace: (_, option) => `if ($self.isEnabled()) ${option} = ${JSON.stringify(RNNOISE_OPTION)};`,
+            }, {
+                match: /(?=\i&&(\i)\.push\(\{name:(?:\i\.){1,2}Messages.USER_SETTINGS_NOISE_CANCELLATION_KRISP)/,
+                replace: (_, options) => `${options}.push({ name: "AI (RNNoise)", value: "${RNNOISE_OPTION}" });`,
+            }, {
+                match: /(?<=onChange:function\((\i)\)\{)(?=(?:\i\.){1,2}setNoiseCancellation)/,
+                replace: (_, option) => `$self.setEnabled(${option}.value === ${JSON.stringify(RNNOISE_OPTION)});`,
+            }],
+        },
+    ],
+
+    setEnabled,
+    isEnabled: () => settings.store.isEnabled,
+    async connectRnnoise(stream: MediaStream): Promise<MediaStream> {
+        if (!settings.store.isEnabled) return stream;
+
+        const audioCtx = new AudioContext();
+        await audioCtx.audioWorklet.addModule(rnnoiseWorkletSrc);
+
+        const rnnoiseWasm = await getRnnoiseWasm();
+        const rnnoise = new RnnoiseWorkletNode(audioCtx, {
+            wasmBinary: rnnoiseWasm,
+            maxChannels: 1,
+        });
+
+        const source = audioCtx.createMediaStreamSource(stream);
+        source.connect(rnnoise);
+
+        const dest = audioCtx.createMediaStreamDestination();
+        rnnoise.connect(dest);
+
+        // Cleanup
+        const onEnded = () => {
+            stream.removeEventListener("inactive", onEnded);
+            rnnoise.disconnect();
+            source.disconnect();
+            audioCtx.close();
+            rnnoise.destroy();
+        };
+        stream.addEventListener("inactive", onEnded);
```

Comment:
```suggestion
        stream.addEventListener("inactive", onEnded, { once: true });
```
then u don't need to remove it

Reply:
ur so cool